Dockerfile-demo

```diff
@@ -0,0 +1,105 @@
+FROM ubuntu:bionic
+
+# install python and dependencies for mesa
```

Comment:
Could you add a description for what mesa is and why you need it? Its taking a considerably long time to build, perhaps we need to create a base image for the core gymfc dependencies and then derive a macos container if that OS needs all these additional software packages.

Reply:
sure, I'll document it.
This is not MacOS related, the docker container is running Ubuntu. This is the most tricky part, enable graphic rendering in a docker container and allow it to forward it to a XHost outside the container.

The pre-installed mesa drives do not support DRI but Gazebo needs them. Hence we need to install new drivers.

Yep, this takes a while. But I build it in a way that the layers can easily be cached by docker, so usually this does not need to be done often. Important is that the ADD/COPY commands come later

Reply:
It was actually documented: _compile mesa to be able to use DRI with docker_

Reply:
I guess what I mean is elaborating a bit more on what mesa is and why its needed (essentially what you were clarifying above to me) so if someone is reading through your docker file and seems all this stuff about mesa they understand why. This will help prevent confusion about why the normal gymfc install doesn't require mesa. It would be helpful to expand the DRI acronym as well. I've dockerized gui apps before and have never mesa, so either briefly explain why the difference or linking to a reference would be helpful to the reader.